numpy/core/src/umath/scalarmath.c.src

```diff
@@ -584,11 +584,15 @@ static NPY_INLINE int
 /**begin repeat
  * #name = byte, short, int, long, longlong#
  * #type = npy_byte, npy_short, npy_int, npy_long, npy_longlong#
+ * #NAME = BYTE, SHORT, INT, LONG, LONGLONG#
  */
 static NPY_INLINE int
 @name@_ctype_absolute(@type@ a, @type@ *out)
 {
     *out = (a < 0 ? -a : a);
+    if (a == NPY_MIN_@NAME@) {
+        return NPY_FPE_OVERFLOW;
+    }
```

Comment:
This will work in practice I guess.  In principle you should likely avoid calculating `-a` since it is undefined behavior in that branch.

Reply:
Went ahead and put this check first, in which case I directly assign `*out = a` for consistent behavior. Let me know if you have any concerns :)